internal/refactoring/move_execute.go

```diff
@@ -28,11 +30,11 @@ import (
 //
 // ApplyMoves expects exclusive access to the given state while it's running.
 // Don't read or write any part of the state structure until ApplyMoves returns.
-func ApplyMoves(stmts []MoveStatement, state *states.State) MoveResults {
+func ApplyMoves(stmts []MoveStatement, state *states.State, providers map[addrs.Provider]providers.Factory) (*states.State, MoveResults, tfdiags.Diagnostics) {
```

Comment:
If we're still modifying the state in place, do we need to return the state argument?

Reply:
Done!